src/test/java/com/jesusgsdev/services/PurchaseServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package com.jesusgsdev.services;
+
+
+import com.fasterxml.jackson.datatype.jdk8.OptionalSerializer;
+import com.jesusgsdev.entities.Book;
+import com.jesusgsdev.entities.Customer;
+import com.jesusgsdev.entities.Purchase;
+import com.jesusgsdev.entities.enums.PaymentMethod;
+import com.jesusgsdev.repositories.PurchaseRepository;
+import org.assertj.core.util.Lists;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+public class PurchaseServiceTest {
+
+    @InjectMocks
+    private PurchaseService purchaseService;
+
+    @Mock
+    private PurchaseRepository purchaseRepository;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+
+    @Test
+    @DisplayName("Add a new find of purchases for customers by email in the Bookshop")
+    public void findPurchasesByCustomerEmailTest() {
+        String email = "johndoe@mailinator.com";
+        //Given
+        Customer customer = new Customer("pedro", "pajares", "avenida palmera45", email);
+        Customer customer2 = new Customer("pedr2o", "pajares2", "avenida palmera452", email);
+        Customer customer3 = new Customer("pedr2o3", "pajares3", "avenida palmera453", email);
+
+        Book book = new Book("ISBN00001", "Book Name", 9.99, "Author Name Test", 200, "provider");
+
+        Optional<PaymentMethod> paymentMethodOptional = PaymentMethod.fromValue("paypal");
```

Comment:
You can just do it in one line:
PaymentMethod paymentMethodOptional = PaymentMethod.fromValue("paypal").get();